Chore/160166795 split gxa test data

alfonsomunozpomer opened the pull request:
It’s been a long journey, but at last, here it is...

# Atlas Core 1.0

I think this is a fully revamped version of the module, thus new name and version. It includes a [git submodule](https://github.com/ebi-gene-expression-group/atlas-schemas) and it runs against the new [test data](https://gitlab-internal.ebi.ac.uk/amunoz/atlas-core-test-data), so I’ve set up a new Bamboo plan where it passes:
http://bar:8085/browse/ATLAS-ATLASCORECI0

The way I envision it, once we merge this it would be a good time to split `gxa` and `sc` to their own repositories and we can have modular deployments without branch hell.

If it helps, we can have a look at it together (or parts of it) if you prefer.

alfonsomunozpomer committed:
Delete all test data

alfonsomunozpomer committed:
Use private as it was ever intended, as a boolean

alfonsomunozpomer committed:
Fix incorrect depth column index, caught by randomized test! :O

alfonsomunozpomer committed:
Reorganize SQL migrations

alfonsomunozpomer committed:
Get rid off experiment_organism, the worst table ever

alfonsomunozpomer committed:
Simplify layout of files

alfonsomunozpomer committed:
Tweak test to adapt it to the new file format, without the term root

alfonsomunozpomer committed:
Rename method, it returns an instance of OntologyTerm

alfonsomunozpomer committed:
Randomize test

alfonsomunozpomer committed:
Migrate InterPro trader to use TsvStreamer
Weirdly enough, CSVReader fails with IPR029684, but with the fantastic TsvStreamer it doesn’t!

alfonsomunozpomer committed:
camelCase all the things

alfonsomunozpomer committed:
Add minimal, basic, sufficient test for ArrayDesign

alfonsomunozpomer committed:
Repackage array design-related classes

alfonsomunozpomer committed:
Flatten package structure (we have just one single “3rd party integration”)

alfonsomunozpomer committed:
Remove forsaken endpoint

alfonsomunozpomer committed:
Do not memoize results, just go to the database as needed; remove method only used in api/arraydesigns.txt endpoint

alfonsomunozpomer committed:
Add some good tests for ArrayDesignDao

alfonsomunozpomer committed:
Feature #160196467 – Expand icon mappings in landing pages (#154)

* Add/remove more species icons based on EBI Icon Font v1.2

* Refactor species icon logic

(cherry picked from commit 739ace2)

alfonsomunozpomer committed:
Make pig warm-blooded instead of photosynthetic (#157)

Hotfix #160200562 – Pig icon should be red

(cherry picked from commit e4193e3)

alfonsomunozpomer committed:
If Comment[AEExperimentDisplayName] is empty default to Investigation Title (#156)

Hotfix #160200567 – E-MTAB-5214 title is empty

(cherry picked from commit e173573)

alfonsomunozpomer committed:
Reformat code; run all JS code after all is loaded; give some size values that work in Firefox *and* Chrome (#155)

Hotfix #160201398 – Tableau is cropped in Chrome

(cherry picked from commit 71711c2)

alfonsomunozpomer committed:
Worms are cold-blooded

alfonsomunozpomer committed:
Another forgotten endpoint bites the dust

alfonsomunozpomer committed:
We don’t need DataFileHubFactory: just injecting the path bean and using DataFileHub’s constructor is enough

alfonsomunozpomer committed:
Rewrite methods so that the ontology terms multimap don’t rely on functions with side effects; be more functional all around

alfonsomunozpomer committed:
Rewrite EfoLookupService as a simpler single class and move it to the conditions package, the only place where it’s used

alfonsomunozpomer committed:
Remove unused method

alfonsomunozpomer committed:
Move out of AssayGroup a convenience constructor used in tests and refactor it into a tests package factory class

alfonsomunozpomer committed:
Add Guava flavour of AssertJ to the project

alfonsomunozpomer committed:
Refactor all uses of ThreadLocalRandom.current() into a private constant

alfonsomunozpomer committed:
Make two conditions equal if the metadata terms are the same, regardless of the order

alfonsomunozpomer committed:
Rename ConditionsLookupService to LegacyConditionsLookupService; refactor design of the class to two classes with only five methods and half the code, without any transient state (i.e. builders), just pure functions

alfonsomunozpomer committed:
Add tests:
• Compare results of legacy service with current class
• Good mocks that mirror our actual data model

alfonsomunozpomer committed:
Remove legacy conditions service

alfonsomunozpomer committed:
Remove all logic that inserts data to the DB from the web app

alfonsomunozpomer committed:
Change varargs to List to get rid of a compilation warning

alfonsomunozpomer committed:
Replace deprecated methods after the template of the SC suggestions without highlighting

alfonsomunozpomer committed:
Minor code clean up

alfonsomunozpomer committed:
Add repository atlas-schemas as a submodule in the test resources directory

alfonsomunozpomer committed:
Add Spring-profile-dependent configuration to use an embedded database when running integration tests (i.e. all tests that use TestConfig for their context configuration)

alfonsomunozpomer committed:
Use PostgreSQL unless other profile such as "test" is set

alfonsomunozpomer committed:
Move fixtures to a separate directory and load them in @BeforeAll for consistency with other ITs, as it’s a more general solution

alfonsomunozpomer committed:
Populate experiments table before running tests

alfonsomunozpomer committed:
Remove PostgreSQL-specific functions
It’s a shame because TABLESAMPLE was faster than random-sorting the results, but since we’ll be using fixtures from now on I don’t think it’s going to be even noticeable

alfonsomunozpomer committed:
Use the already-defined transaction manager bean from our JdbcConfig class

alfonsomunozpomer committed:
It’s not organism, but species!

alfonsomunozpomer committed:
Rewrite test suite to include: parameterized tests, embedded DB and SQL fixtures

alfonsomunozpomer committed:
Add H2 dependency and remove PostgreSQL connection parameters from the integration profile

alfonsomunozpomer committed:
Load all the test configuration from TestConfig instead of segregating beans in two files
This is really a workaround for the fact that @ActiveProfiles is skipped in configuration classes and only works for tests. For instance, you can have a base integration test class that includes this annotation and make all other ITs inherit it.

alfonsomunozpomer committed:
Split fixture files into data insertion and table clean up
Unfortunately @Sql doesn’t work with @ParameterizedTest :(

alfonsomunozpomer committed:
Ditch Spring profiles and load/skip the necessary @Configuration with filters in @ComponentScan
We need to set the active profile at compile time (through environment or -D variables) so that the right @Configuration classes are loaded when @ComponentScan is found, but then we can’t set a profile when running Tomcat. In that case we’d need to activate a profile programmatically in AppConfig or similar. In summary: it’d be a mess.

alfonsomunozpomer committed:
Name returned columns explicitly and don’t rely on PostgreSQL-specifics

alfonsomunozpomer committed:
Add schemas submodule directory entry

alfonsomunozpomer committed:
Clean up tables after array design-related classes have been tested

alfonsomunozpomer committed:
Comment where the fixtures come from

alfonsomunozpomer committed:
Test PopularSpeciesDao with sensible-looking data

alfonsomunozpomer committed:
Wrap id-to-EFO map in a LazyReference to speed up application start up
This is how it was before, now it’s time to bring it back again now that everything seems to be working fine

alfonsomunozpomer committed:
A new beginning...

pcm32 commented:
Kudos for separate repositories!

monicajianu commented:
@alfonsomunozpomer It would've been great if the code linting bits of this PR were part of a separate branch. And I'm sure other parts could've been split up in more manageable chunks. As it is, 3209 files have been changed which makes it a little hard to review!

monicajianu commented on base/src/main/java/uk/ac/ebi/atlas/species/services/PopularSpeciesDao.java:
```diff
@@ -21,7 +21,7 @@
             "FROM experiment " +
             "LEFT OUTER JOIN  experiment_organism ON experiment_organism.experiment=experiment.accession " +
-            "WHERE private='F' GROUP BY experiment.type, experiment_organism.organism";
+            "WHERE private=FALSE GROUP BY experiment.type, experiment_organism.organism";
```
These changes will require updating every existing experiment in the staging/prod databases, I'm guessing. Something to keep in mind.

monicajianu reviewed (changes requested):
What are the steps needed to be done locally so that the application runs properly? As it stands, I cannot run any test or run the applications at all due to various errors popping up everywhere.
I hava already run 
``` 
git submodule init
git submodule update
```

alfonsomunozpomer commented:
@monicajianu Unfortunately just deleting the test data affects over 3,000 files. I think that the sane way to review this PR is by checking each commit, that’s why I said that you might want to have a look at it with another pair of eyeballs.

Also, yes, the way to get this running is:
```
git submodule init
git submodule update

cd <somewhere_else>
git clone --recurse-submodules git@gitlab-internal.ebi.ac.uk:amunoz/atlas-core-test-data.git
```

The argumen `--recurse-submodules` is equivalent to `git clone`, `submodule init` and `submodule update`. You might need to enter your password a few times since the submodules URLs use HTTPS, I noticed this later and will change to SSH but as it is now it’s a bit annoying.

And lastly:
```
ln -s <somewhere_else> <location_of_integration_test_data>
```

You should move your old copy of `integration_test_data`, so that when you switch back to the old branch you don’t need to do all the `ln`s.

monicajianu commented:
@alfonsomunozpomer It looks like you also need to add `atlas-core` to the sc and gxa `pom.xml` files? Otherwise you get lots of dodgy unknown imports.

alfonsomunozpomer commented:
Yes. I didn’t mention it because at this stage it’s only this module what needs to be double checked. Interoperability with modules `gxa` and `sc` will be comming soon.

monicajianu commented on base/src/test/java/uk/ac/ebi/atlas/bioentity/interpro/InterProTSVParserTest.java:
```diff
@@ -1,81 +1,69 @@
+@MockitoSettings(strictness = Strictness.LENIENT)
+class InterProTSVParserTest {
     private static final String IPR000001 = "IPR000001";
     private static final String KRINGLE = "Kringle";
```
Apparently a Kringle is a type of Scandinavian pastry! 😄 Methinks a biologist was hungry when they came up with this particular concept.

monicajianu commented on base/src/test/java/uk/ac/ebi/atlas/model/ArrayDesignTest.java:
```diff
@@ -0,0 +1,14 @@
+    @Test
+    void unknownArrayDesignsArePopulatedWithTheirAccession() {
+        assertThat(ArrayDesign.createForUnknownName(randomAlphanumeric(20)))
+                .satisfies(arrayDesign -> assertThat(arrayDesign.accession()).isEqualTo(arrayDesign.name()));
```
While I appreciate that AssertJ has the capability for this `satisfies` thing, I'm not sure it helps readability (or brevity)...

monicajianu commented on gxa/src/main/java/uk/ac/ebi/atlas/controllers/rest/api/AssayDesignController.java:
```diff
@@ -1,34 +0,0 @@
-
-@Controller
-@Scope("request")
-public class AssayDesignController {
```
What was this used for?

alfonsomunozpomer replied:
Maria used it for something on her pipeline. Sorry but I can’t remember the specifics.

monicajianu commented on gxa/src/main/java/uk/ac/ebi/atlas/ebeye/DifferentialExperimentContrastsTsvController.java:
```diff
@@ -1,79 +0,0 @@
-package uk.ac.ebi.atlas.ebeye;
```
We are still providing some sort of data for ebeye, aren't we? Is that coming straight from data prod?

alfonsomunozpomer replied:
There are still two services we provide:
• Endpoint `/das/s4/features`
• The widget

monicajianu commented on base/src/main/java/uk/ac/ebi/atlas/model/experiment/ExperimentDesign.java:
```diff
@@ -150,46 +141,39 @@ public void addAssayHeader(String assayHeader) {
+                        entry ->
+                                entry.getValue().values().stream()
+                                        .flatMap(sampleCharacteristic ->
+                                                sampleCharacteristic.valueOntologyTerms().stream())
```
Due to `SampleCharacteristic` using `AutoValue`, it means that the naming convention is ever so slightly different. You have `sampleCharacteristic.valueOntologyTerms()` but you have `factor.*get*ValueOntologyTerms()`. Besides that these getters are pretty much identical... Some inheritance and refactoring the models might simplify things a bit.

monicajianu replied:
Ahh okay, I just noticed the `FactorSet` complication. Nevermind... The whole thing could do with some heavy re-engineering...

monicajianu commented on base/src/test/java/uk/ac/ebi/atlas/experimentimport/analyticsindex/conditions/ConditionsCollectorTest.java:
```diff
@@ -0,0 +1,279 @@
+    @Test
+    void differentialConditions() {
+        assertThat(subject.getConditions(differentialExperimentMock))
+//                .isEqualTo(legacyConditionsLookupService.buildPropertiesForDifferentialExperiment(differentialExperimentMock.getAccession(), experimentDesignMock, ImmutableList.of(contrast)))
```
Leftover commented out line

monicajianu commented on base/src/test/java/uk/ac/ebi/atlas/experimentimport/analyticsindex/conditions/ConditionsCollectorTest.java:
```diff
@@ -0,0 +1,279 @@
+    private static <K, V> Multimap<K, V> randomizedMultimapOf(Collection<K> buckets, Collection<V> values) {
+        checkArgument(values.size() >= buckets.size());
+
+        List<K> _buckets = Lists.newArrayList(buckets);
```
What does the underscore in `_variableName` meant to represent...?

alfonsomunozpomer replied:
It’s a way to indicate that it holds a (modified) copy of one of the parameters.

alfonsomunozpomer committed:
Remove commented out line

monicajianu commented on base/src/main/java/uk/ac/ebi/atlas/experimentimport/ExperimentCrud.java:
```diff
@@ -89,11 +84,6 @@ public UUID importExperiment(String experimentAccession, boolean isPrivate) thro
         }
 
-        // We only insert in the DB differential and single cell experiments expressions
-        analyticsLoaderFactory
```
Does data prod have data loading scripts for Expression Atlas yet...?

alfonsomunozpomer replied:
No, not yet. From the webapp we only need to load coexpressions and index in Solr.

monicajianu commented on base/src/main/java/uk/ac/ebi/atlas/experimentimport/ExperimentCrud.java:
```diff
@@ -122,11 +112,6 @@ public UUID importSingleCellExperiment(String experimentAccession, boolean isPri
         }
 
-        // We only insert in the DB differential and single cell experiments expressions
-        analyticsLoaderFactory
```
👍

monicajianu commented on gxa/src/main/java/uk/ac/ebi/atlas/experimentpage/ExperimentPageService.java:
```diff
@@ -88,7 +93,7 @@ private JsonElement experimentDescription(Experiment experiment,
     }
 
-    private URI callbackLink(String urlBase, String accessKey, Pair<String, String>... params) {
+    private URI callbackLink(String urlBase, String accessKey, List<Pair<String, String>> params) {
```
Why does this method need to take a list? It is only ever called with a single Pair or no pair. So an `Optional<Pair>`?

monicajianu commented on base/src/main/java/uk/ac/ebi/atlas/search/suggester/SuggesterService.java:
```diff
@@ -22,10 +28,24 @@ public SuggesterService(SuggesterDao suggesterDao, SpeciesFactory speciesFactory
+                .map(SUGGESTION_TO_MAP);
+    }
+
+    public Stream<Map<String, String>> fetchPropertiesWithHighlighting(String query, String...  species) {
```
Could you maybe add a comment here saying what the difference between highlighting and no highlighting is? Because there is nothing in the return type that suggests a difference, and it is a very Solr-specific behaviour.

alfonsomunozpomer replied:
Sounds sensible.

monicajianu commented on base/src/main/java/uk/ac/ebi/atlas/search/suggester/SuggesterService.java:
```diff
@@ -1,18 +1,24 @@
-    private static final int DEFAULT_MAX_NUMBER_OF_SUGGESTIONS = 20;
+    public static final int DEFAULT_MAX_NUMBER_OF_SUGGESTIONS = 20;
+
+    private static final Function<Suggestion, Map<String, String>>
```
Hmm adding `Suggestion` here couples this service to Solr, when it fact it should be the Dao who is concerned about where the suggestions are coming from.

alfonsomunozpomer replied:
I agree in principle. The reason why I didn’t do it was that, after looking at [Solr’s `Suggestion` class](https://github.com/apache/lucene-solr/blob/1d85cd783863f75cea133fb9c452302214165a4d/solr/solrj/src/java/org/apache/solr/client/solrj/response/Suggestion.java), it seemed a bit superfluous to create pretty much the same thing under our own namespace. It’s just a data class with no behaviour.

monicajianu commented on base/src/test/java/uk/ac/ebi/atlas/model/arraydesign/ArrayDesignDaoIT.java:
```diff
@@ -38,7 +43,16 @@
+            "SELECT identifier, COUNT(DISTINCT(designelement)) AS count FROM designelement_mapping GROUP BY identifier";
+
+    @BeforeAll
+    void beforeAllTests() {
```
I'm going to be a bit picky here and say that this method name is a bit repetitive and not very descriptive, if you read it together with the annotation. Something like `populateDatabase` would read a bit better IMO.

alfonsomunozpomer replied:
You’re totally right.

alfonsomunozpomer committed:
Add comment to clarify the difference between highlighted and non-highlighted suggestions

alfonsomunozpomer committed:
Be descriptive about what @BeforeAll and @AfterAll actually do

alfonsomunozpomer committed:
Don’t build the assay-to-conditions map every time we visit an assay, instead build it once and reuse it when building the stream elements
Try to ANALYTICS_IMPORT GTEx without this... I did and it stalls for an indefinite amount of time until it starts to send documents to Solr. I waited for three hours until I decided something was wrong.

alfonsomunozpomer committed:
Resotre .gitignore file, it got lost somewhere along the way

alfonsomunozpomer committed:
Add additional external properties for ZK host and Solr port

alfonsomunozpomer committed:
Only strip descriptions with [Sourece:...], as there are other descriptions with interesting stuff in square brackets; parameterize tests, JUnit 5

alfonsomunozpomer committed:
Update comments to be truthful; remove unnecessary throw declaration

alfonsomunozpomer committed:
Remove unused constant

alfonsomunozpomer committed:
Rename generic type to be consistent across classes

alfonsomunozpomer committed:
Lower log level, it floods the logger when adding documents to the analytics index and performance really suffers

alfonsomunozpomer committed:
Add a public rawQuery method
This is a small defeat in our mini SorlCloud framework since by exposing this some type-safety is lost, but now we can write better tests. We can rethink the design if we move the framework to its own project.

alfonsomunozpomer committed:
Remove any mention of Spring profiles... ever

alfonsomunozpomer committed:
Add scxa-gene2experiment to the list of required collections

alfonsomunozpomer committed:
Modify some of the expected results to match our new test data

alfonsomunozpomer committed:
Add/modify settings for the new test data

alfonsomunozpomer committed:
Fix path of CI test data

alfonsomunozpomer committed:
Use version number with patch level

alfonsomunozpomer committed:
Remove redundant .gitignore file
I think this one was accidentally moved in a previous commit (?)

alfonsomunozpomer committed:
Remove redundant .gitignore file
I think this one was accidentally moved in a previous commit (?)

alfonsomunozpomer committed:
Restructure app configuration classes to exclude Postgres-related beans in test classes; add new Solr/ZK port property

alfonsomunozpomer committed:
Restructure app configuration classes to exclude Postgres-related beans in test classes; add new Solr/ZK port property

alfonsomunozpomer committed:
Change test configuration class to TestConfig instead of WebConfig

alfonsomunozpomer committed:
Change test configuration class to TestConfig instead of WebConfig

alfonsomunozpomer committed:
Add SQL fixtures to tests, migrate to JUnit 5 and parameterize

alfonsomunozpomer committed:
Add SQL fixtures to tests, migrate to JUnit 5 and parameterize

alfonsomunozpomer committed:
Inline JSON response in test

alfonsomunozpomer committed:
Inline JSON response in test

alfonsomunozpomer committed:
Fix RSE to work with the new schema that uses a BOOLEAN column instead of a VARCHAR

alfonsomunozpomer committed:
Fix RSE to work with the new schema that uses a BOOLEAN column instead of a VARCHAR

alfonsomunozpomer committed:
Use new suggestion field names

alfonsomunozpomer committed:
Use new suggestion field names

alfonsomunozpomer committed:
Wrap IOException in UncheckedIOException

alfonsomunozpomer committed:
Wrap IOException in UncheckedIOException

alfonsomunozpomer committed:
Remove superfluous test cases
Import operations just insert a row in the experiments test, there’s no longer a different behaviour depending on the experiment type, so we only need to test the operations regardless of the type

alfonsomunozpomer committed:
Remove superfluous test cases
Import operations just insert a row in the experiments test, there’s no longer a different behaviour depending on the experiment type, so we only need to test the operations regardless of the type

alfonsomunozpomer committed:
Move to JUnit 5 and leave the tables clean after testing

alfonsomunozpomer committed:
Move to JUnit 5 and leave the tables clean after testing

alfonsomunozpomer committed:
By default return public experiments

alfonsomunozpomer committed:
By default return public experiments

alfonsomunozpomer commented:
@monicajianu Could you please review the last set of changes since your approval?

alfonsomunozpomer committed:
Restructure app configuration classes to exclude Postgres-related beans in test classes; add new Solr/ZK port property

alfonsomunozpomer committed:
Change test configuration class to TestConfig instead of WebConfig

alfonsomunozpomer committed:
Replace SQL fixtures with a set that spans all tables used by SCEA and have tests use them

alfonsomunozpomer committed:
Fix tests and add a new helper method to JdbcUtils to make tests a bit less like a real scenario, but that’s as far as fixtures will let us go

alfonsomunozpomer committed:
Script to create fixtures for SCEA

monicajianu commented on base/src/main/java/uk/ac/ebi/atlas/configuration/SolrConfig.java:
```diff
@@ -15,18 +15,21 @@
+    public CloudSolrClient cloudSolrClient(@Value("${zk.host}") String zkHost,
+                                           @Value("${zk.port}") String zkPort) {
+        return new CloudSolrClient.Builder().withZkHost(zkHost + ":" + zkPort).build();
 //        return SOLR_CLIENT_CACHE.getCloudSolrClient(zkHost);
```
Is commented out line still needed?

alfonsomunozpomer replied:
Removed. Good catch.

alfonsomunozpomer committed:
Remove commented out getter
We don’t want to use the cache this way because it might not be populated

monicajianu commented on base/src/main/java/uk/ac/ebi/atlas/experimentimport/analyticsindex/conditions/ConditionsCollector.java:
```diff
@@ -36,22 +34,33 @@ public ConditionsCollector(EfoLookupService efoLookupService) {
 
     // Visits all assays in each factor and builds-collects the conditions in a stream
     public Stream<Condition> getConditions(BaselineExperiment experiment) {
+        ImmutableSetMultimap<String, String> assayAccession2OntologyTerms =
```
This should probably go in a separate (private) function to avoid code duplication: lines 60-63 seem identical to this block, unless I'm missing something...

alfonsomunozpomer replied:
This is the usual compromise on DRY vs “flow” (for lack of a better word). In this case I thought that adding a level of indirection wouldn’t add value to code readability, but now I’m not so sure. I’ve added a commit to change this. Let me know if you think this reads better.

monicajianu commented on pom.xml:
```diff
@@ -632,8 +634,13 @@
                 <jdbc.url>jdbc:postgresql:gxpatlasloc</jdbc.url>
                 <jdbc.username>atlas3dev</jdbc.username>
                 <jdbc.password>atlas3dev</jdbc.password>
+                <!-- If you’re running a Beowu^W k8s cluster with minikube, the host of ZK and Solr will be given by
```
Errr... what's a `Beowu^W`?

alfonsomunozpomer replied:
Ah... [kids these days!](https://en.wikipedia.org/wiki/Talk:Beowulf_cluster#What_Is_Funny_About_/._Jokes?)

monicajianu replied:
😕

alfonsomunozpomer replied:
Also: https://en.wikipedia.org/wiki/Backspace#^H

Old jokes from old people. ;)

monicajianu commented on base/src/main/java/uk/ac/ebi/atlas/testutils/SolrUtils.java:
```diff
@@ -1,36 +1,65 @@
 
-    private SolrCloudCollectionProxyFactory solrCloudCollectionProxyFactory;
+    private AnalyticsCollectionProxy analyticsCollectionProxy;
+    // private BioentitiesCollectionProxy bioentitiesCollectionProxy;
```
More lonely commented out lines ☹️

alfonsomunozpomer replied:
Fixed. There was also a method which was no longer used (the no-args random gene ID discussed below).

monicajianu commented on base/src/main/java/uk/ac/ebi/atlas/testutils/SolrUtils.java:
```diff
@@ -1,36 +1,65 @@
                 .toString();
     }
+
+    public String fetchRandomGeneIdFromAnalytics(AnalyticsSchemaField field, String term) {
```
I'm not sure this method name or its parameters are descriptive enough about what is going on... The method is basically getting a random gene ID that satisfies a condition, but you wouldn't know what without looking at the contents.

alfonsomunozpomer replied:
The fact that you pass a field and a value is a strong enough hint, IMO. Take, for instance, [ThreadLocalRandom.nextInt(int origin, int bound)](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadLocalRandom.html#nextInt-int-int-); any random generator with arguments imposes certain conditions or restrictions on the returned value. I can add a comment if you want, but I think it’s quite straightforward.

monicajianu commented:
A couple of very minor things, but on the whole it looks good! Re-approved!

alfonsomunozpomer committed:
DRY out some code that was doing the same in both baseline and differential experiments

alfonsomunozpomer committed:
Remove unused code and comments

alfonsomunozpomer committed:
Change test name to better indicate its scope

alfonsomunozpomer committed:
Change test name to better indicate its scope

alfonsomunozpomer committed:
Make test a bit more robust and meaningful, just by checking that there’s at least one Solr document

alfonsomunozpomer committed:
Make test a bit more robust and meaningful, just by checking that there’s at least one Solr document

alfonsomunozpomer committed:
Remove unnecessary `andReturn()` method calls

alfonsomunozpomer committed:
Remove unnecessary `andReturn()` method calls

monicajianu committed:
Update sc/src/test/java/uk/ac/ebi/atlas/search/geneids/GeneIdSearchDaoIT.java

monicajianu committed:
Update sc/src/test/java/uk/ac/ebi/atlas/search/geneids/GeneIdSearchDaoIT.java

alfonsomunozpomer committed:
Merge pull request #178 from ebi-gene-expression-group/chore/160166795-split-gxa-test-data-sc-module

Use H2 DB and new test data in ITs in sc module

alfonsomunozpomer committed:
Merge branch 'chore/160166795-split-gxa-test-data' into chore/160166795-split-gxa-test-data-gxa-module

alfonsomunozpomer committed:
Merge pull request #179 from ebi-gene-expression-group/chore/160166795-split-gxa-test-data-gxa-module

Weird artifact PR for merging on another PR

lingyun1010 committed:
Delete local images in release-notes

lingyun1010 committed:
Show release notes from github page with iframe

alfonsomunozpomer committed:
Add iFrame-Resizer to dynamically expand release notes iframe

alfonsomunozpomer committed:
Put actually-used method back

alfonsomunozpomer committed:
Remove odd tab

alfonsomunozpomer committed:
Merge remote-tracking branch 'origin/dev' into chore/160166795-split-gxa-test-data

alfonsomunozpomer committed:
Post-merge fixes

alfonsomunozpomer committed:
Add submodules
